tests/test_regression_ler_decrease.py

```diff
@@ -0,0 +1,195 @@
+        ],
+    )
+    def test_depolarizing_noise(self, code_gen, decoder_cls, p, measured_op):
+        """Test for depolarizing noise."""
+        sizes = np.arange(start=5, stop=10, step=2)
+        lers = np.zeros(np.size(sizes))
+
+        for i, size in enumerate(sizes):
+            ssc = code_gen(int(size))
+            lers[i] = get_logical_error_rate(
+                ssc,
+                error_data=noise_model(ssc, qubit_p=p),
+                rounds=size + 1,
+                reps=10000,
+                logical_op=measured_op,
+                decoder_cls=decoder_cls,
+            )
+
+        assert all(x >= y for x, y in zip(lers[:-1], lers[1:])), (
+            f"Numbers are not in decreasing order. \nLERs: {lers}%",
+        )
+
+    @pt.mark.slow()
+    @pt.mark.parametrize(
+        "decoder_cls", [decoders.PyMatchingDecoder, decoders.FusionBlossomDecoder]
+    )
+    @pt.mark.parametrize(
+        "code_gen", [codes.Code.make_planar, codes.Code.make_rotated_planar]
+    )
+    @pt.mark.parametrize(
+        "depol_p, meas_p, measured_op", [(0.01, 0.01, "X"), (0.01, 0.01, "Z")]
+    )
+    def test_depolarizing_with_measurement_noise(
+        self, code_gen, decoder_cls, depol_p, meas_p, measured_op
+    ):
+        """Test for depolarizing and measurement noise."""
+        sizes = np.arange(start=5, stop=10, step=2)
+        lers = np.zeros(np.size(sizes))
+
+        for i, size in enumerate(sizes):
+            ssc = code_gen(int(size))
+            lers[i] = get_logical_error_rate(
+                ssc,
+                error_data=noise_model(ssc, qubit_p=depol_p, meas_p=meas_p),
+                rounds=size + 1,
+                reps=10000,
+                logical_op=measured_op,
+                decoder_cls=decoder_cls,
+            )
+
+        assert all(x >= y for x, y in zip(lers[:-1], lers[1:])), (
+            f"Numbers are not in decreasing order. \nLERs: {lers}%",
+        )
+
+    @pt.mark.slow()
+    @pt.mark.parametrize(
+        "decoder_cls", [decoders.PyMatchingDecoder, decoders.FusionBlossomDecoder]
+    )
+    @pt.mark.parametrize(
+        "code_gen",
+        [
+            codes.Code.make_planar,
+            codes.Code.make_rotated_planar,
+            codes.Code.make_subsystem_surface_code,
+        ],
+    )
+    @pt.mark.parametrize(
+        "depol_p, meas_p, gate_p, measured_op",
+        [(0.01, 0.01, 0.01, "X"), (0.01, 0.01, 0.01, "Z")],
+    )
+    def test_depolarizing_gate_measurement_noise(
+        self, code_gen, decoder_cls, depol_p, meas_p, gate_p, measured_op
+    ):
+        """Test for depolarizing, measurement and gate noise."""
+        sizes = np.arange(start=5, stop=10, step=2)
+        lers = np.zeros(np.size(sizes))
+
+        for i, size in enumerate(sizes):
+            ssc = code_gen(int(size))
+            lers[i] = get_logical_error_rate(
+                ssc,
+                error_data=noise_model(ssc, qubit_p=depol_p, meas_p=meas_p),
```

Comment:
@isolatedinformation I think that `gate_p` needs to be passed here into `noise_model`
```suggestion
                error_data=noise_model(ssc, qubit_p=depol_p, meas_p=meas_p, gate_p=gate_p),
```